fluent.runtime/tests/format/test_attributes.py

```diff
@@ -0,0 +1,149 @@
+        val, errs = self.ctx.format('ref-bar', {})
+        self.assertEqual(val, 'Bar Attribute')
+        self.assertEqual(len(errs), 0)
+
+    def test_can_be_formatted_directly_for_entities_with_string_values(self):
+        val, errs = self.ctx.format('foo.attr', {})
+        self.assertEqual(val, 'Foo Attribute')
+        self.assertEqual(len(errs), 0)
+
+    def test_can_be_formatted_directly_for_entities_with_pattern_values(self):
+        val, errs = self.ctx.format('bar.attr', {})
+        self.assertEqual(val, 'Bar Attribute')
+        self.assertEqual(len(errs), 0)
+
+
+class TestAttributesWithSimplePatternValues(unittest.TestCase):
+
+    def setUp(self):
+        self.ctx = FluentBundle(['en-US'], use_isolating=False)
+        self.ctx.add_messages(dedent_ftl("""
+            foo = Foo
+            bar = Bar
+                .attr = { foo } Attribute
+            baz = { foo } Baz
+                .attr = { foo } Attribute
+            qux = Qux
+                .attr = { qux } Attribute
+            ref-bar = { bar.attr }
+            ref-baz = { baz.attr }
+            ref-qux = { qux.attr }
+        """))
+
+    def test_can_be_referenced_for_entities_with_string_values(self):
+        val, errs = self.ctx.format('ref-bar', {})
+        self.assertEqual(val, 'Foo Attribute')
+        self.assertEqual(len(errs), 0)
+
+    def test_can_be_formatted_directly_for_entities_with_string_values(self):
+        val, errs = self.ctx.format('bar.attr', {})
+        self.assertEqual(val, 'Foo Attribute')
+        self.assertEqual(len(errs), 0)
+
+    def test_can_be_referenced_for_entities_with_pattern_values(self):
+        val, errs = self.ctx.format('ref-baz', {})
+        self.assertEqual(val, 'Foo Attribute')
+        self.assertEqual(len(errs), 0)
+
+    def test_can_be_formatted_directly_for_entities_with_pattern_values(self):
+        val, errs = self.ctx.format('baz.attr', {})
+        self.assertEqual(val, 'Foo Attribute')
+        self.assertEqual(len(errs), 0)
+
+    def test_works_with_self_references(self):
+        val, errs = self.ctx.format('ref-qux', {})
+        self.assertEqual(val, 'Qux Attribute')
+        self.assertEqual(len(errs), 0)
+
+    def test_works_with_self_references_direct(self):
+        val, errs = self.ctx.format('qux.attr', {})
+        self.assertEqual(val, 'Qux Attribute')
+        self.assertEqual(len(errs), 0)
+
+
+class TestMissing(unittest.TestCase):
+    def setUp(self):
+        self.ctx = FluentBundle(['en-US'], use_isolating=False)
+        self.ctx.add_messages(dedent_ftl("""
+            foo = Foo
+            bar = Bar
+                .attr = Bar Attribute
+            baz = { foo } Baz
+            qux = { foo } Qux
+                .attr = Qux Attribute
+            ref-foo = { foo.missing }
+            ref-bar = { bar.missing }
+            ref-baz = { baz.missing }
+            ref-qux = { qux.missing }
+            attr-only
+                     .attr  = Attr Only Attribute
```

Comment:
There are still a few messages across tests which use the old, equal-sign-less syntax for messages which only define attributes. Would you mind fixing them before this PR is merged, please?